File: backend/src/cli/options.ts
import process from 'node:process'
import { PKG_NAME, VERSION } from '@frontend/enum/version.ts'
import { internalConfig } from '@/config/config'

export interface CliOverrides {
  port?: string
  host?: string
  dataDir?: string
  createConfig: boolean
  withTls: boolean
  noTui: boolean
  help: boolean
  version: boolean
}

export function defaultCliOverrides(): CliOverrides {
  return {
    createConfig: false,
    withTls: false,
    noTui: false,
    help: false,
    version: false,
  }
}

/** Must run before loadConfig. */
export function applyDataDirOverride(overrides: CliOverrides) {
  if (overrides.dataDir) {
    process.env.FILE_LITE_DATA_BASE_DIR = overrides.dataDir
  }
}

/** CLI flags > env > config.json. Call after loadConfig. */
export function applyCliOverrides(overrides: CliOverrides) {
  if (!internalConfig.config) {
    return
  }
  if (overrides.port !== undefined) {
    internalConfig.config.port = overrides.port
  }
  if (overrides.host !== undefined) {
    internalConfig.config.host = overrides.host
  }
}

export function printVersion() {
  console.log(`${PKG_NAME} v${VERSION}`)
}

export function printHelp() {
  console.log(`${PKG_NAME} v${VERSION}

Usage:
  file-lite [options]

Options:
  -h, --help           Show help
  -v, --version        Show version
  --no-tui             Run without interactive menu
  --create-config      Create config.json if missing and exit
  --with-tls           With --create-config: generate self-signed cert via openssl
  -p, --port <port>    Override listen port
  -H, --host <host>    Override listen host
  --data-dir <path>    Data directory (default: ./file-lite under cwd;
                       env: FILE_LITE_DATA_BASE_DIR)

Ephemeral mode (no config.json): no files are written; use printed Ticket to sign in.
`)
}

File: backend-go/cli/options.go
package cli

import (
	"fmt"
	"os"

	"file-lite-go/config"
)

type Overrides struct {
	Port         string
	Host         string
	DataDir      string
	CreateConfig bool
	WithTLS      bool
	TLSHosts     []string
	NoTui        bool
	Help         bool
	Version      bool
}

func ApplyDataDirOverride(o Overrides) {
	if o.DataDir != "" {
		_ = os.Setenv("FILE_LITE_DATA_BASE_DIR", o.DataDir)
	}
}

func ApplyCliOverrides(o Overrides) {
	config.ApplyListenOverrides(o.Port, o.Host)
}

func PrintVersion() {
	fmt.Printf("%s v%s\n", config.PkgName, config.Version)
}

func PrintHelp() {
	fmt.Printf(`%s v%s

Usage:
  file-lite-go [options]

Options:
  -h, --help             Show help
  -v, --version          Show version
  --no-tui               Run without interactive menu
  --create-config        Create config.json and exit
    --with-tls           Also generate self-signed cert
      --tls-host <host>  Add domain or IP to the cert (repeatable)
  -p, --port <port>      Override listen port
  -H, --host <host>      Override listen host
  --data-dir <path>      Data directory (default: ./file-lite under cwd;
                         env: FILE_LITE_DATA_BASE_DIR)

Ephemeral mode (no config.json): no files are written; use printed Ticket to sign in.
`, config.PkgName, config.Version)
}
